commons/src/tracing.rs

```diff
@@ -0,0 +1,32 @@
+//! Tracing service.
+
+use opentelemetry::api::{Key, Provider};
+use opentelemetry::{global, sdk};
+use opentelemetry_jaeger::{Exporter, Process};
+
+/// init_tracer sets up Jaeger tracer
+pub fn init_tracer(name: &'static str) -> thrift::Result<()> {
+    let exporter = Exporter::builder()
+        .with_agent_endpoint("127.0.0.1:6831".parse().unwrap())
+        .with_process(Process {
+            service_name: name.to_string(),
+            tags: vec![Key::new("exporter").string("jaeger")],
+        })
+        .init()?;
+
+    let provider = sdk::Provider::builder()
+        .with_simple_exporter(exporter)
+        .with_config(sdk::Config {
+            default_sampler: Box::new(sdk::Sampler::Always),
+            ..Default::default()
+        })
+        .build();
+    global::set_provider(provider);
+
+    Ok(())
+}
+
+/// get_tracer returns an instance of global tracer
+pub fn get_tracer() -> global::BoxedTracer {
+    global::trace_provider().get_tracer("")
```

Comment:
>it wraps just a simple, single line of code, and it leaks the external library's type

it does wrap a single line of code, but it gets used often. Should I create a macros instead?

As for the type I'm not too much concerned really - we still have to import external type's traits and can't fully abstract from a particular library anyway

>By using the same name (the empty string) for all callers, this will return the same tracer for all calls

It returns the default tracer, which is the one created with "init_tracer". It would be distinct in graph-builder and policy-engine.

Reply:
> it does wrap a single line of code, but it gets used often. Should I create a macros instead?

I see, I think it's fine. Optimization advocates would maybe suggest to put a `#[inline(always)]` on top of this function, but I trust the compiler to make this decision.



> As for the type I'm not too much concerned really - we still have to import external type's traits and can't fully abstract from a particular library anyway

I think we can, but I understand you're implying that we don't want to.

Reply:
I don't think abstracting too much is a good idea. We already get all the features we want now from currently selected libraries, so I don't anticipate a switch to a different set of libs anytime soon